### DISCUSSION TITLE
I want to make it so that when using `->searchable()`, relations that have been deleted cannot be selected.

### DISCUSSION THREAD

**Discussion by bzy107** [filamentphp/filament#9271] [Help]
The `patients` table has an `owner_id` column. 
In the `owners` table, let's assume I can delete data related to a specific patients by its ID.

In this scenario, 
I want the `owner_id` to be displayed on the `Patients` form, but I don't want the associated data to be displayed. 
I'm not sure how to customize this.

Here is the relevant code where `->searchable()` is used, and the `owner_id` is displayed as is:

```php
->relationship('owner', 'name')
->searchable()
->preload()
```

The image shows the display when the Owner table with id: 3 has been deleted.
![スクリーンショット 2023-10-24 2 44 09](https://github.com/filamentphp/filament/assets/28666304/4be9d5fd-3079-40e3-b854-ad1eafd21291)

**Comment by ArtMin96** [comment]
[Customizing the relationship option labels](https://filamentphp.com/docs/3.x/forms/fields/checkbox-list#customizing-the-relationship-option-labels)

**Reply by bzy107** [reply]
thanks. 
Even with the changes made, the ID is still being displayed. 
```php
                    ->relationship(
                        name: 'owner',
                        titleAttribute: 'name',
                        modifyQueryUsing: fn (Builder $query) => $query->where('id', 1)
                    )
```
![スクリーンショット 2023-10-24 3 26 28](https://github.com/filamentphp/filament/assets/28666304/f7da327e-7d9d-46fe-8f0c-bf56004b9631)



I believe the solution might require adjustments in the `->searchable()` method rather than the `->relationship()`.
However, I haven't found a solution yet.

```php
->relationship('owner', 'name')
// ->searchable()
->preload()
```
![スクリーンショット 2023-10-24 3 28 33](https://github.com/filamentphp/filament/assets/28666304/d70caf48-3dfa-4fc6-893e-843857b59cfb)

**Reply by ArtMin96** [reply]
I mean `->getOptionLabelFromRecordUsing(fn (Model $record)`

**Reply by bzy107** [reply]
I tried using getOptionLabelFromRecordUsing, but the issue persisted.

**Reply by leandrocfe** [reply]
You shouldn't allow deleting items if they are being used.
https://laravel.com/docs/10.x/migrations#foreign-key-constraints

`$table->nullOnDelete()` can be a workaround.

**Reply by bzy107** [reply]
Thank you. Your argument is correct.
I was trying to handle an exceptional case, so I will close this discussion.
